pkg/databind/internal/secrets/obfuscate.go

```diff
@@ -0,0 +1,90 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package secrets
+
+import (
+	"encoding/base64"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/newrelic/infrastructure-agent/pkg/databind/pkg/data"
+)
+
+type Obfuscated struct {
+	Key    string `yaml:"key"`
+	Secret string `yaml:"secret"`
+}
+
+type obfuscateGatherer struct {
+	cfg *Obfuscated
+}
+
+// ObfuscateGatherer instantiates a Obfuscate variable gatherer from the given configuration. The fetching process
+// will return either a map containing access paths to the stored JSON.
+// E.g. if the stored Secret is `{"account":{"user":"test1","password":"test2"}}`, the returned Map
+// contents will be:
+// "account.user"     -> "test1"
+// "account.password" -> "test2"
+func ObfuscateGatherer(obfuscated *Obfuscated) func() (interface{}, error) {
+	g := obfuscateGatherer{cfg: obfuscated}
+	return func() (interface{}, error) {
+		dt, err := g.get()
+		if err != nil {
+			return "", err
+		}
+		return dt, err
+	}
+}
+
+func (g *obfuscateGatherer) get() (data.InterfaceMap, error) {
+	credentials := g.cfg
+
+	decrypted, err := decryptStringWithKey(credentials.Secret, credentials.Key)
+	if err != nil {
+		return nil, fmt.Errorf("failed to decode obfuscated secret: %v", err)
+	}
+	result := data.InterfaceMap{}
+	if err := json.Unmarshal([]byte(decrypted), &result); err != nil {
+		result["password"] = decrypted
```

Comment:
If the text isn't JSON how about just returning the decrypted string instead instead of map with the key `password`? The KMS does something similar to this and so it makes the config nicer as if you know it is just one value you don't have to remember to do `{something.password}` and can use `{something}`.